src/pybind/mgr/dashboard/services/auth/oauth2.py

```diff
@@ -106,6 +125,7 @@ def _create_user(cls):
             user = mgr.ACCESS_CTRL_DB.create_user(
                 jwt_payload['sub'], None, jwt_payload['name'], jwt_payload['email'])
         except UserAlreadyExists:
+            logger.debug("User already exists")
             user = mgr.ACCESS_CTRL_DB.get_user(jwt_payload['sub'])
```

Comment:
When user exists, should we check the roles and update them if they changed?

Reply:
Yeah the next line does that and I tested this scenario as well.
`user.set_roles(cls.get_user_roles())`